plugins/slicer/MONAILabel/MONAILabelLib/client.py

```diff
@@ -168,7 +169,8 @@ def download_label(self, label_id, tag):
                 MONAILabelError.SERVER_ERROR, f"Status: {status}; Response: {response}", status, response
             )
 
-        local_filename = tempfile.NamedTemporaryFile(dir=self._tmpdir, suffix=".nii.gz").name
+        file_ext = slicer.util.settingsValue("MONAILabel/fileExtension", ".nii.gz")
```

Comment:
This is OK (an improvement over using a hardcoded extension), but it is not very robust. The setting is shared between multiple Slicer instances that are running at the same time, which may interfere with each other. For example, a user may start some lengthy batch processing operation with one MONAILabel server and in the meantime working on some other data sets interactively on another MONAILabel server, using another Slicer instance. If they use different file formats then one of them may break.

Instead, it would be much more robust and not much more complicated to get the file extension from the file content. It is trivial for NRRD and DICOM:
- NRRD: first 4 bytes of the file is `NRRD`
- DICOM: 4 characters at position 128 are `DICM`

(For NIFTI, as usual, things are more complicated, there is no easy or efficient way to tell that if a file is a NIFTI image. It may be a gzipped file then you need to start with unzipping, then get the "magic string" from the header and then compare it to a list of known valid values. The Python reader that you already use can probably check this, but it may not be efficient - e.g., if it unzipped the entire file for this check then that would be probably an unacceptable overhead).

It would also make sense for the server to provide a filename with a correct extension (for example via `filename` in the `Content-Disposition` header).